multichain-testing/test/tools/wallet.test.ts

```diff
@@ -0,0 +1,44 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { makeQueryClient } from '../../tools/query.js';
+import { createWallet } from '../../tools/wallet.js';
+import { sleep } from '../../tools/sleep.js';
+import { commonSetup } from '../support.js';
+
+const test = anyTest as TestFn<Record<string, never>>;
+
+test('create a wallet and get tokens', async t => {
+  const { useChain } = await commonSetup(t);
+
+  const prefix = useChain('osmosis').chain.bech32_prefix;
+  const wallet = await createWallet(prefix);
+  const addr = (await wallet.getAccounts())[0].address;
+  t.regex(addr, /^osmo1/);
+  t.log('Made temp wallet:', addr);
+
+  const apiUrl = useChain('osmosis').getRestEndpoint();
+  const queryClient = makeQueryClient(apiUrl);
+  t.log('Made query client');
+
+  const { balances } = await queryClient.queryBalances(addr);
+  t.log('Beginning balances:', balances);
+  t.deepEqual(balances, []);
+
+  const osmosisFaucet = useChain('osmosis').creditFromFaucet;
+  t.log('Requeting faucet funds');
+  await osmosisFaucet(addr);
+  await sleep(1000, t.log); // needed to avoid race condition
+
+  const { balances: updatedBalances } = await queryClient.queryBalances(addr);
+  t.like(updatedBalances, [{ denom: 'uosmo', amount: '10000000000' }]);
+  t.log('Updated balances:', updatedBalances);
+
+  const bondDenom =
+    useChain('osmosis').chain.staking?.staking_tokens?.[0].denom;
+  if (!bondDenom) {
+    t.fail('Bond denom not found.');
```

Comment:
```suggestion
  t.truthy(bondDenom, 'Bond denom not found.');
```
would be more clear. I imagine you used the `if` to get type narrowing for the `queryBalance` call. This is a reason to adopt Ava 6. IIUC its assertions also do type assertions so lines after `t.truthy`will know `bondDenom` is not nullish.

Reply:
> Ava 6. IIUC its assertions also do type assertions so lines after t.truthywill know bondDenom is not nullish.

No luck here, seemingly.  I used the [non-null-assertion-operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) to make this more readable and drop the `if/else`

Reply:
confirmed, though I can't make sense of it. There clearly is a type assertion.
<img width="1042" alt="Screenshot 2024-06-12 at 1 07 05 PM" src="https://github.com/Agoric/agoric-sdk/assets/21505/0feed051-18cd-4f9f-b43e-eebaf1fbf226">